backend/service/topology/cache.go

```diff
@@ -27,22 +31,22 @@ func (c *client) acquireTopologyCacheLock(ctx context.Context) {
 	}
 	defer conn.Close()
 
-	advisoryLockId := convertLockIdToAdvisoryLockId(topologyCacheLockId)
+	go func() {
+		<-ctx.Done()
+		ticker.Stop()
+		c.unlockAdvisoryLock(context.TODO(), conn, advisoryLockId)
```

Comment:
```suggestion
		c.unlockAdvisoryLock(context.Background(), conn, advisoryLockId)
```

TODO should only be used in the case that we plan to later plumb a different context or something.